Tests/SonarScanner.MSBuild.Test/RollForwardTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SonarScanner.MSBuild.Test
+{
+    [TestClass]
+    public class RollForwardTests
+    {
+        [TestMethod]
+        public void MustRollForward()
+        {
+            var scannerCsProjRelativePath = Path.Combine("src", "SonarScanner.MSBuild", "SonarScanner.MSBuild.csproj");
+            // for dev environment
+            var scannerCsProjPath = Path.Combine("..", "..", "..", "..", "..", scannerCsProjRelativePath);
+            if (!File.Exists(scannerCsProjPath))
+            {
+                // CI environment
+                var dir = Environment.GetEnvironmentVariable("BUILD_SOURCESDIRECTORY");
+                Assert.IsNotNull(dir);
```

Comment:
It would helpful when diagnosing failures if we could easily tell from the logs whether the failure is due to the test not being able to locate the file to check (an issue with the test), or because there is an issue in the product code is wrong.
If a failure is due to a test issue, I add an assertion message starting with something like "[TEST ERROR]" e.g. "[TEST ERROR] Failed to locate the file to check".